workspaces/src/error/impls.rs

```diff
@@ -35,7 +35,10 @@ impl Error {
         }
     }
 
-    pub(crate) fn full<T, E>(kind: ErrorKind, msg: T, error: E) -> Self
+    /// Construct a workspaces [`Error`] with the full details of an error which includes
+    /// the internal error it references, the custom error message with further context
+    /// and the [`ErrorKind`] that represents the category of error.
+    pub fn full<T, E>(kind: ErrorKind, msg: T, error: E) -> Self
```

Comment:
Just going to share some low-confidence opinions that having the semantics of error and message separate seems unnecessarily complex and having this interface now be public seems like it might be a pain to maintain in the future.

If you're confident this is useful and these should all be exposed, go for making this change, I just noticed that there is no actual usage of the downcasting through `into_inner` of `Error` and exposing all of these variants will make it even harder to switch later.

Reply:
I see, how about just expose only custom/message errors then. There's only a couple cases where full is useful in workspaces itself where we just provide some extra context about the error. So let's not expose it for now until there's a community need